### ISSUE TITLE
Pop-up dialog behavior

### ISSUE THREAD

**Issue by Kurios** [MegaMek/mekhq#102]
Started this as an issue before I actually started working on this.

Currently, popup elements (and other visual elements) are stored as variable in the CampaignGUI class. This means they hold onto resources, even when they are not rendered. They "pop up" when the element is set to visible. However, the advantage (and disadvantage) to this is the elements do not need to be recomputed with they are set to visible. (The disadvantage here is they do not necessarily refresh on pop up because of this.)

The other way of doing it, recreating the element on pop-up, would save some memory (Didn't really check how much), and allow us to have multiple pop-ups of the same type. (With the current method, there is only one, and it is either visible or not.)

I believe the memory savings may be worthwhile as there are some people who do not like my approach to handling memory in megamek/mekhq (Throw 16GB at it) and the ability to have more then one pop-up could be a good bonus.

**TLDR: I'm proposing to convert all pop up elements so they are removed from memory on close, and created on open.**

**Comment by pheonixstorm** [comment]
Not everyone has 16Gb to throw at it either. I do like the idea though. Get it done so I can check it out :)

**Comment by sixlettervariables** [comment]
As of 0.45.5-SNAPSHOT there really are only a few dialogs that are persistent in CampaignGUI:
- HistoricalDailyReportDialog
- AdvanceDaysDialog
- BloodnameDialog

This is unlikely to cause large amounts of memory usage, but in the interest of cleaning up unnecessary fields I'll make those locals.

**Comment by sixlettervariables** [comment]
The memory savings landed in #1221.
